src/core/tileLayer.js

```diff
@@ -920,9 +912,31 @@
 
           // mark the tile as covered
           this._setTileTree(tile);
-        }.bind(this));
+        } else {
+          tile.then(function () {
+            if (tile !== this.cache.get(tile.toString())) {
+              /* If the tile has fallen out of the cache, don't draw it -- it
```

Comment:
No firm numbers, but some anecdotal results.

Using the vgl renderer, we render no more often then the browser animation frame rate.  Before this change, when panning slowly and consistently, it seemed like we missed nearly half of those frames on my desk top machine, largely due to time spend in GC.  After this change, I only see a missed frame during a slow pan when we add six or more tiles concurrently.